src/lava/magma/compiler/compiler_graphs.py

```diff
@@ -242,6 +241,11 @@ def is_dag(self, graph: 'DiGraphBase' = None) -> ty.Tuple[bool,
             _, graph = self.is_dag(graph)
 
         if len(list(graph.nodes)) > 1:
+            isolated_nodes = \
+                [node for node, nodeattr in graph.nodes.items() if nodeattr[
+                    'degdef'] == NodeAnnotation.ISOLATED]
+            if len(list(graph.nodes)) == len(isolated_nodes):
+                return True, graph
```

Comment:
With the new definition of `leaves`, I don't think we will ever run out of nodes by pruning everything out completely. So, we will never get `len(list(graph.nodes)) == 0` and never use the last `return`.

So, I think we don't need three `return` statements. But let's keep them for now.